pkg/controller/pxcbackup/controller.go

```diff
@@ -129,57 +128,114 @@ func (r *ReconcilePerconaXtraDBClusterBackup) Reconcile(ctx context.Context, req
 		return reconcile.Result{}, err
 	}
 
+	err = r.ensureFinalizers(ctx, cr)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "ensure finalizers")
+	}
+
 	err = r.tryRunBackupFinalizers(ctx, cr)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrap(err, "failed to run finalizers")
+		return reconcile.Result{}, errors.Wrap(err, "run finalizers")
 	}
 
-	if cr.Status.State == api.BackupSucceeded ||
-		cr.Status.State == api.BackupFailed {
+	if cr.Status.State == api.BackupSucceeded || cr.Status.State == api.BackupFailed {
 		if len(cr.GetFinalizers()) > 0 {
 			return rr, nil
 		}
+
 		return reconcile.Result{}, nil
 	}
 
 	if cr.DeletionTimestamp != nil {
 		return rr, nil
 	}
 
-	cluster, err := r.getCluster(cr)
+	passedSeconds, err := r.checkPassiveDeadline(ctx, cr)
```

Comment:
Since we are returning the `passedSeconds ` only to log them afterwards and we don't have another use case for them, maybe it is better to not return them at all and move the log inside the function.